src/firewall/core/nftables.py

```diff
@@ -238,17 +239,37 @@ def __init__(self, fw):
         self._nft_handle.set_handle_output(True)
         self._nft_handle.set_json_output(True)
 
-    def nft_cmd(self, json_root):
+    def nft_cmd(self, json_root, is_large=False):
         if isinstance(json_root, list):
             # The caller gave us the JSON in a list, so we can drop the
             # reference to the memory. Unpack the list, and delete the entry.
             l = json_root
             (json_root,) = l
             del l[0]
-        json_root_str = json.dumps(json_root)
-        del json_root
-        json_root_str = json_root_str.encode("utf-8")
-        rc, output, error = self._nft_handle.cmd(json_root_str)
+
+        if is_large and hasattr(self._nft_handle, "cmd_from_file"):
+            filename = None
+            try:
+                with firewall.functions.tempFile(
+                    prefix="temp.nft-json.", delete=False
+                ) as file:
+                    filename = file.name
+                    json.dump(json_root, file)
+
+                del json_root
+
+                rc, output, error = self._nft_handle.cmd_from_file(
+                    filename.encode("utf-8")
+                )
+            finally:
+                if filename is not None:
+                    firewall.functions.removeFile(filename)
+        else:
+            json_root_str = json.dumps(json_root)
+            del json_root
+            json_root_str = json_root_str.encode("utf-8")
+            rc, output, error = self._nft_handle.cmd(json_root_str)
```

Comment:
Instead of adding complexity by sometimes doing it directly and sometimes doing it via a file, why not just always do it from a file?

Reply:
I think it will be slower to write a small file. The branch change should not make some cases slower.